minio/xml.py

```diff
@@ -80,20 +92,43 @@ def findtext(element, name, strict=False):
     return element.text or ""
 
 
-def unmarshal(cls, xmlstring):
+K = TypeVar("K")
+
+
+class IFromXML(Protocol):
+    """typing stub for class with `fromxml` method"""
+
+    @classmethod
+    def fromxml(cls: Type[K], element: ET.Element) -> K:
+        """Create python object with values from XML element."""
+
+
+T = TypeVar("T", bound=IFromXML)
+
+
+def unmarshal(cls: Type[T], xmlstring: str) -> T:
     """Unmarshal given XML string to an object of passed class."""
     return cls.fromxml(ET.fromstring(xmlstring))
 
 
-def getbytes(element):
+def getbytes(element: ET.Element) -> bytes:
     """Convert ElementTree.Element to bytes."""
-    data = io.BytesIO()
-    ET.ElementTree(element).write(
-        data, encoding=None, xml_declaration=False,
-    )
-    return data.getvalue()
+    with io.BytesIO() as data:
+        ET.ElementTree(element).write(
+            data,
+            encoding=None,
+            xml_declaration=False,
+        )
+        return data.getvalue()
+
+
+class IToXML(Protocol):
```

Comment:
How about using abstract class instead of `Protocol` inheritance?

Reply:
That would involve many other files

Reply:
I couldn't able to understand why it involves other files.

Reply:
abstract class doesn't work like `Protocol` in typing, abstract class is expected to be inherited as base class, but in typing `Protocol` mean duck typing.

This mean, when a class support `toxml(...)`, this class need to use `IToXMl` as base class, which dould involve other files.

Reply:
example:
```python
import abc
from abc import ABC


class ABCToXML(ABC):

    @abc.abstractmethod
    def toxml(self, element: str | None) -> str:
        ...


class ImplCorrect(ABCToXML):
    """"""

    def toxml(self, element: str | None) -> str:
        return ""


class ConsiderNotImplemented:
    """"""

    def toxml(self, element: str | None) -> str:
        return ""


_1: ABCToXML = ImplCorrect()
_2: ABCToXML = ConsiderNotImplemented() # type error here
```